jina/docker/hubio.py

```diff
@@ -309,15 +309,18 @@ def build(self) -> Dict:
 
             with TimeContext(f'building {colored(self.args.path, "green")}', self.logger) as tc:
                 try:
-                    self._check_completeness()
+                    _check_result = self._check_completeness()
                     self._freeze_jina_version()
 
+                    _dockerfile_path = _check_result['Dockerfile']
```

Comment:
So I think here it should be `_dockerfile_path = 'Dockerfile'`, or alternatively something like `_check_result['Dockerfile'].split('/')[-1]` (to be better prepared for different dockerfile names, such as the one for GPU).

This is because in the call to the client below, `path=self.args.path` is passed, which means that the `dockerfile` location will be relative to that `path`. However in `_check_completness` the dockerfile path is obtained by joining `self.args.path` and `Dockerfile`. I think this was the reason that #1732 broke the hub builds.

Reply:
Also, back then the issue was that CI didn't catch this error for some reason. I hope now it will

Reply:
hmm, the tests actually work locally. I've put them into the integration tests and let's see how ci says.

Reply:
Aaaah I figured it out! 

The reason test worked and hub build broke was because in tests `args.path` is specified with an absolute path - for example `hub_build_test` uses `os.path.join(cur_dir, 'hub-mwu')`, where `cur_dir` is an absolute path, while building in Jina Hub is done using relative paths, e.g. `encoders/audio/MFCCTimbreEncoder/`. 

Tests worked absolutely fine, but they were only done for the absolute path - where both `path` and `dockerfile` are absolute paths. If both are relative, then docker SDK tries to join them together, which doesn't work unless `dockerfile` is really just the name of the file inside `args.path`.

So yes, extracting the path using `os.path.basename` as you are doing now is the right way (btw you have to do it in another location in the file, that's why the CI is failing). All that should be added is a test where `args.path` is passed as a relative path, and we are good to go :D

Reply:
Great catch! Added another test on the relative pathes.